### DISCUSSION TITLE
Shared Pages

### DISCUSSION THREAD

**Discussion by Sen-Gupta** [oqtane/oqtane.framework#4816] [Ideas]
Hello @sbwalker,

I have my eyes on Oqtane for last 3 + years and I love the way it is growing.

I'm in a process to recommend this to one of our project which will leverages tenancy.
A Master site like www.mytest.com having some public pages and a few private admin pages. This tenant will have sites like www.mytest.com/site1 and www.mytest.com/site2.

We expect to have more than 100 sites. All these sites will have a few shared pages other than their own site specific pages. Every time, if we update layout of any of the shared pages, it should reflect on all sites.

Now, Looking at SiteTemplate and the way pages are constructed from modules, and module data with page definition is stored in database.

Even if we update SiteTemplate, it only reflects changes on new sites being created that point onwards but I couldn't find a way to update all sites. This is an issue for the client as they can't go and update all sites.

If we add/update a new page in SiteTemplate it should appear on all sites based on SiteTemplate.

How to achieve this?

Nice meeting you again!
Sen

**Comment by sbwalker** [comment]
@Sen-Gupta SiteTemplates were designed to be used as part of creating new sites... they were not intended for the scenario you are describing (ie. updating existing sites). You should investigate the ISiteMigration interface instead. 

https://www.oqtane.org/blog/!/38/site-migrations-deployment

**Comment by Sen-Gupta** [comment]
@sbwalker thanks. I understand, now we need to approach the problem a bit differently.
